app/src/main/java/fr/free/nrw/commons/category/CategoriesModel.java

```diff
@@ -121,8 +124,8 @@ boolean cacheContainsKey(String term) {
         }
 
         //otherwise, search API for matching categories
-        return mwApi
-                .allCategories(term, SEARCH_CATS_LIMIT)
+        return categoryClient
```

Comment:
`searchCategories` and `allcategories` return different kind of responses. Here we are supposed to use `allcategories` as it searches based on prefix. 

For eg if we search for `christchurch` using both the APIs the response would be different. 

Eg for `allcategories`: https://commons.wikimedia.org/w/api.php?aclimit=25&action=query&format=json&list=allcategories&acprefix=christchurch

Eg for `searchCategories`: https://commons.wikimedia.org/w/api.php?action=query&format=json&formatversion=2&generator=search&gsrnamespace=14&gsrsearch=christchurch&gsrlimit=25

Reply:
This replicates the prefix behaviour we desire but doesn't sort 
items alphabetically. I am stuck with using generator because that's what Wikimedia Dataclient supports. Is that ok? 

https://commons.wikimedia.org/w/api.php?action=query&format=json&formatversion=2&generator=allcategories&gacprefix=christchurch&gaclimit=25

Reply:
Yes, i think that's perfectly fine. We can probably sort the result manually.

Reply:
Done, thanks for catching that :)